subcommands/targets/edit.go

```diff
@@ -36,7 +36,7 @@ func doEdit(cmd *cobra.Command, args []string) {
 	// Get raw json
 	targets, err := api.TargetsList(factory)
 	subcommands.DieNotNil(err)
-	orig, err := json.MarshalIndent(targets.Signed.Targets, "", "  ")
+	orig, err := json.MarshalIndent(targets, "", "  ")
```

Comment:
that's not going to work. I think `TargetList` was returning the exact targets content. Now its returning your representation of it? The edit command needs the *exact* content so that it can edit things correctly.

Reply:
I'm not sure I understand what does _exact content_ mean in this context.
Before my change it was returning _[]tuf.Files_ wrapped into a _SignedTargets_, not it returns unwrapped _[]tuf.Files_, which is the same as before, there is no such thing as _my representation_ in this command.

There is _my representation_ for waves, but it's an internal struct inside a function, so that nobody can use it anywhere else.

Reply:
I use a slightly different representation of signed targets for waves (because of an issue with delegations), but even there it's still the same _[]tuf.Files_;  moreover, that's an inner struct inside a _doCreateWave_ function i.e. cannot be used elsewhere.

Reply:
the old TargetsList was calling:

 https://api.foundries.io/ota/repo/lmp/api/v1/user_repo/targets.json

Your change has it calling:

  https://api.foundries.io/ota/factories/lmp/targets/

Are these really the *exact* same data? I thought your new view had small differences?

Reply:
The only difference is in a wrapping envelope:
- old data was like this:
  `signatures: <signatures>
   signed:
      _type: Targets
      expires: <expires>
      targets: <targets-list>
      version: <version>
  `
- new data is like this:
  `<targets-list>`
But it is the same `<targets-list>` in both cases, and `fioctl targets edit` doesn't use the other data, it only uses a list of targets.  To what I know at least a canonical form of them must be _exact_ same data.

Reply:
ack. thanks